src/mono/mono/mini/monovm.c

```diff
@@ -228,45 +228,40 @@ monovm_initialize (int propertyCount, const char **propertyKeys, const char **pr
 	return 0;
 }
 
-void mono_extract_and_register_properties (const char *buffer, long fileSize);
+void mono_extract_and_register_properties (const char *buffer, guint64 file_size);
 
 // Initialize monovm with properties set by runtimeconfig.json. Primarily used by mobile targets.
 int
 monovm_runtimeconfig_initialize (MonovmRuntimeConfigArguments *arg, MonovmRuntimeConfigArgumentsCleanup cleanup_fn, void *user_data)
 {
 	switch (arg->kind) {
 	case 0: {
-		FILE *filePtr = NULL;
-		long fileSize = 0;
-		char *buffer = NULL;
-		size_t result;
-
-		filePtr = fopen(arg->runtimeconfig.name.path, "rb");
-		g_assert (filePtr != NULL);
-
-		fseek (filePtr , 0 , SEEK_END);
-		fileSize = ftell (filePtr);
-		rewind (filePtr);
-
-		if (fileSize <= 0)
+		guint64 file_size = 0;
+		MonoFileMap *map;
+		void *ret_handle;
+		char *buffer;
+
+		map = mono_file_map_open (arg->runtimeconfig.name.path);
+		g_assert (map);
+		file_size = mono_file_map_size (map);
+		buffer = (char*)mono_file_map (file_size, MONO_MMAP_READ|MONO_MMAP_PRIVATE, mono_file_map_fd (map), 0, &ret_handle);
+		g_assert (buffer);
+
+		if (file_size <= 0)
 			return 0;
 
-		buffer = (char *) malloc (sizeof (char) * fileSize);
-		g_assert (buffer != NULL);
-
-		result = fread (buffer,1,fileSize,filePtr);
-		g_assert (result == fileSize);
-
-		mono_extract_and_register_properties (buffer, fileSize);
+		mono_extract_and_register_properties (buffer, file_size);
```

Comment:
There's no need to do this early.  Do it in `mono_runtime_install_appctx_properties` - you can delay calling the cleanup_fn until `_install` - this data will stick around.

Reply:
Moved.